conbench/app/_plots.py

```diff
@@ -85,28 +89,49 @@ def get_history_plot(
 
         # The number (1, 2, 3?) maybe needs to be tuned further. Also see
         # https://github.com/conbench/conbench/issues/867
-        if len(samples) > 2:
-            assert isinstance(samples[0], HistorySample)
-            jsondoc = json.dumps(
-                bokeh.embed.json_item(
-                    time_series_plot(
-                        samples=samples,
-                        current_benchmark_result=benchmark_result,
-                        run=run,
-                    ),
-                    f"plot-history-{i}",  # type: ignore
+        if len(samples) < 3:
+            # This reason/error will be shown verbatim in HTML, so this should be
+            # a nice message.
+            return BokehPlotJSONOrError(
+                None,
+                f"not enough history items yet ({len(samples)}). Keep submitting "
+                "results for this specific benchmark scenario (case permutation, "
+                "and context)!",
+            )
+
+        highlight_other: Optional[Tuple[HistorySample, str]] = None
+
+        if highlight_other_result is not None:
+            s_by_id: Dict[str, HistorySample] = {
+                s.benchmark_result_id: s for s in samples
+            }
+
+            if highlight_other_result.bmrid not in s_by_id:
+                return BokehPlotJSONOrError(
+                    None,
+                    "you asked to highlight the benchmark result with ID "
+                    f"<kbd>{highlight_other_result.bmrid}</kbd> in this view. "
+                    "However, that result is not part of the history of the "
+                    f"result <kbd>{current_benchmark_result.id}</kbd>",
```

Comment:
kbd is styled by bootstrap
https://getbootstrap.com/docs/5.2/content/reboot/#user-input